app/services/course/assessment/submission/update_service.rb

```diff
@@ -224,4 +232,8 @@ def valid_for_grading?(answer)
 
     answer.specific.attempting_times_left > 0 || can?(:manage, @assessment)
   end
+
+  def monitoring_service
```

Comment:
I think we can load monitoring_service before action in submissions_controller. Then we can just call `@monitoring_service` here without reinitializing again.

Reply:
At first, I was going to ask if services are allowed to cross-interact like that.

But then I saw the definition of `initialize` in `Course::Assessment::Submission::UpdateService` which basically takes all the instance variables from the controller and redefine them in it. Personally, I don't think this is right, but since it's already like this, there's no need to re-initialise `Course::Assessment::Submission::MonitoringService` here.